src/handlers/shared/get-user-role.ts

```diff
@@ -0,0 +1,32 @@
+import { Context } from "../../types";
+
+interface MatchingUserProps {
+  role: string;
+  limit: number;
+}
+
+export async function getUserRole(context: Context, user: string): Promise<MatchingUserProps> {
+  const orgLogin = context.payload.organization?.login;
+  const { config, logger } = context;
+  const { maxConcurrentTasks } = config.miscellaneous;
+
+  try {
+    const response = await context.octokit.orgs.getMembershipForUser({
+      org: orgLogin as string,
+      username: user,
+    });
+
+    const matchingUser = maxConcurrentTasks.find(({ role }) => role.toLowerCase() === response.data.role);
+
+    if (matchingUser) {
+      //chech if the current user role matches any of those defined in the config
+      return matchingUser;
+    } else {
+      //return the role with the smallest task limit
+      return maxConcurrentTasks.reduce((minTask, currentTask) => (currentTask.limit < minTask.limit ? currentTask : minTask));
```

Comment:
We could move this outside of the try block so that we always return something from this function instead of throwing an error, which is the intent of this line anyway.

Reply:
Makes sense!